remixer_storage/src/main/java/com/google/android/libraries/remixer/storage/json/StoredVariable.java

```diff
@@ -0,0 +1,141 @@
+  /**
+   * The RemixerItem's key.
+   */
+  String key;
+  /**
+   * The RemixerItem's title. It's optional.
+   */
+  String title;
+  /**
+   * The data type this variable represents, it's the string representation of one of the {@link
+   * SupportedDataType}.
+   */
+  String dataType;
+
+  // From here, this only applies to Variables, not triggers.
+  /**
+   * The currently selected value for the variable.
+   */
+  T selectedValue;
+  /**
+   * The list of possible values, if this is not null then this StoredVariable represents a {@link
+   * com.google.android.libraries.remixer.ItemListVariable}.
+   */
+  List<T> possibleValues;
+
+  // These are only used in case it T is a number and it is a range variable, otherwise they are
+  // ignored.
+  /**
+   * The minimum value for the {@link com.google.android.libraries.remixer.RangeVariable}
+   */
+  T minValue;
+  /**
+   * The maximum value for the {@link com.google.android.libraries.remixer.RangeVariable}
+   */
+  T maxValue;
+  /**
+   * The increment value for the {@link com.google.android.libraries.remixer.RangeVariable}
+   */
+  T increment;
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+
+    StoredVariable<?> variable = (StoredVariable<?>) o;
+
+    if (!key.equals(variable.key)) return false;
+    if (title != null ? !title.equals(variable.title) : variable.title != null) return false;
+    if (!dataType.equals(variable.dataType)) return false;
+    if (selectedValue != null ?
+        !selectedValue.equals(variable.selectedValue) : variable.selectedValue != null)
+      return false;
+    if (possibleValues != null ?
+        !possibleValues.equals(variable.possibleValues) : variable.possibleValues != null)
+      return false;
+    if (minValue != null ? !minValue.equals(variable.minValue) : variable.minValue != null)
+      return false;
+    if (maxValue != null ? !maxValue.equals(variable.maxValue) : variable.maxValue != null)
+      return false;
+    return increment != null ? increment.equals(variable.increment) : variable.increment == null;
+  }
+
+  @Override
+  public int hashCode() {
+    int result = key.hashCode();
+    result = 31 * result + (title != null ? title.hashCode() : 0);
+    result = 31 * result + dataType.hashCode();
+    result = 31 * result + (selectedValue != null ? selectedValue.hashCode() : 0);
+    result = 31 * result + (possibleValues != null ? possibleValues.hashCode() : 0);
+    result = 31 * result + (minValue != null ? minValue.hashCode() : 0);
+    result = 31 * result + (maxValue != null ? maxValue.hashCode() : 0);
+    result = 31 * result + (increment != null ? increment.hashCode() : 0);
+    return result;
+  }
+
+  /**
+   * Creates a Stored variable from a existing RemixerItem.
+   */
+  static StoredVariable fromRemixerItem(RemixerItem item) {
+    StoredVariable storage = null;
```

Comment:
You define `StoredVariable variable` elsewhere which I think is a better name than `storage`